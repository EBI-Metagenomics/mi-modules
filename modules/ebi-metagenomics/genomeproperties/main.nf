
process GENOMEPROPERTIES {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container 'microbiome-informatics/genome-properties:2.0'

    input:
    tuple val(meta), path(ips)

    output:
    tuple val(meta), path("*.txt")  , emit: summary
    tuple val(meta), path("*.json") , emit: json
    tuple val(meta), path("*.tsv")  , emit: tsv
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def gp_version = "2.0" // No way to get the version from the tool directly so have to hardcode

    """
    assign_genome_properties.pl \\
        ${args} \\
        -matches ${ips} \\
        -gpdir /opt/genome-properties/flatfiles/ \\
        -gpff genomeProperties.txt \\
        -name ${prefix} \\

    mv JSON_${prefix} ${prefix}_gp.json
    mv SUMMARY_FILE_${prefix} ${prefix}_gp.txt
    mv TABLE_${prefix} ${prefix}_gp.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genomeproperties: ${gp_version}
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def gp_version = "2.0" // No way to get the version from the tool directly so have to hardcode

    """
    touch ${prefix}_gp.json
    touch ${prefix}_gp.txt
    touch ${prefix}_gp.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genomeproperties: ${gp_version}
    END_VERSIONS
    """
}
