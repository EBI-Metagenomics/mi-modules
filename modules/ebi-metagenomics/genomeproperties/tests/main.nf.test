
nextflow_process {

    name "Test Process GENOMEPROPERTIES"
    script "../main.nf"
    process "GENOMEPROPERTIES"
    config "./nextflow.config"


    tag "modules"
    tag "modules_nfcore"
    tag "genomeproperties"

    test("genome-properties - interproscan tsv") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_ips.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.summary).match("summary") }
            )
        }

    }

    test("genome-properties - interproscan tsv - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/test_ips.tsv", checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
