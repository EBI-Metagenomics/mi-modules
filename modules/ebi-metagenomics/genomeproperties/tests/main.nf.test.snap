{
    "summary": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_gp.txt:md5,6be55888c673b2b534e70cfb68dd8d84"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-06-21T16:43:45.739990611"
    },
    "genome-properties - interproscan tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,de75b3812fee791338e6848ea7b8a23a"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_gp.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de75b3812fee791338e6848ea7b8a23a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.3"
        },
        "timestamp": "2024-06-21T15:24:59.78129995"
    }
}